pycparser/c_parser.py

```diff
@@ -783,6 +783,12 @@ def p_declaration_specifiers_no_type_3(self, p):
         """
         p[0] = self._add_declaration_specifier(p[2], p[1], 'function')
 
+    # This is a bit ugly, but we need to process atomic specifier before qualifiers,
```

Comment:
Not sure what "a bit ugly" refers to here. Can you clarify the comment?

Reply:
This group of functions has `no_type` in the list, but this code is exactly adding `type`.

Reply:
Maybe it should actually be `qual` here.